source/common/http/async_client_impl.cc

```diff
@@ -267,7 +267,7 @@ void AsyncRequestImpl::onComplete() {
                                                    response_->trailers(), streamInfo(),
                                                    Tracing::EgressConfig::get());
 
-  callbacks_.onSuccess(std::move(response_));
+  callbacks_.onRequestSuccess(this, std::move(response_));
```

Comment:
if we're always passing `this` i don't think it can be null? should we be passing a reference instead?

Reply:
fixed